src/allocator/memserver_allocator.cpp

```diff
@@ -107,6 +129,30 @@ void Memserver_Allocator::reset_profile() {
     MEMSERVER_PROFILE_START_TIME(NVMM)
 }
 void Memserver_Allocator::dump_profile() { NVMM_PROFILE_DUMP(); }
+
+void Memserver_Allocator::garbage_collector_th(uint64_t thread_index)
+{
+    //sleep(1);
+    gc_th_struct_t *gc_th_obj = &delayed_free_thread_array[thread_index];
+    while(gc_th_obj->pthread_running == true) 
+    {
+        if(!gc_th_obj->heap_list.empty())
+	{
+            std::list<gc_heap_struct_t *> heap_list_p = gc_th_obj->heap_list;
```

Comment:
You need to take thread read/read-write locks, Or else it will crash when region is destroyed.

Reply:
gc_th_obj is not the global one. So I dont think we need a lock here.